### ISSUE TITLE
Support working directory of process

### ISSUE THREAD

**Issue by CodingFabian** [oshi/oshi#272]
I am evaluating Oshi as a replacement for sigar. So far I am super excited!

Sigar has an API ProcExe to get cwd, the current working directory of a process.
I looked into Oshis Process classes but could not find such a feature. Therefore I am asking somebody to implement this.

Depending on our progress on the migration I might end up here implementing this myself.

**Comment by dbwiddis** [comment]
We'd love if you can implement this yourself and submit a PR.

**Comment by dbwiddis** [comment]
I originally thought this would be trivial to implement (and it is in other OS's) but it's not quite in Windows.  So whether this goes into 3.4.0 depends on how much time I have, although @CodingFabian could certainly do a PR for it ;)

On Windows we need to use the process handle that was used for Group / Group ID and add more code, [see here](http://stackoverflow.com/questions/14018280/how-to-get-a-process-working-dir-on-windows).

On macOS it's in the [pid_info](http://stackoverflow.com/questions/8327139/working-directory-of-running-process-on-mac-os) structure we already fetch.

On Linux it's dynamically linked by `/proc/<pid>/cwd`.

On Unix `pwdx <pid>` works. The `lsof` command may allow getting multiple processes with a single command line (search for lines containing `cwd`).

**Comment by dbwiddis** [comment]
I spent the better part of my free time trying to implement this in Windows, and it simply doesn't work.  The only place the information is available is in the [Process Environment Block (PEB)](https://msdn.microsoft.com/library/aa813706(VS.85).aspx), which has different structures on 32 bit and 64 bit. The limited info shown has this disclaimer:
> [This structure may be altered in future versions of Windows.]

To access the PEB requires the [NtQueryInformationProcess](https://msdn.microsoft.com/en-us/library/ms684280(v=vs.85).aspx) function which says:
> The NtQueryInformationProcess function and the structures that it returns are internal to the operating system and subject to change from one release of Windows to another. To maintain the compatibility of your application, it is better to use public functions mentioned in the description of the ProcessInformationClass parameter instead.

Of course, these functions are for other info; there is nothing for the working directory.

There are a few C-based projects out there that have done it, but they are individually compiled, and are still complex.  There's different approaches for 64-bit programs calling 32-bit processes (special structures) or 32-bit programs calling 64-bit processes (special functions), plus the normal difference in 32 and 64 bit structures.  And they all are using "undocumented" code per above.

On top of all this, this only would work with a user with admin rights, and the oshi program run as an administrator.

The simplest approach would be to simply command-line parse a downloaded program from Microsoft that does the hard work.  Like [Handle](https://technet.microsoft.com/en-us/sysinternals/bb896655.aspx) or [TList](https://msdn.microsoft.com/en-us/library/windows/hardware/ff558903(v=vs.85).aspx).  But it's absurd to expect most people to download those programs.

Bottom line -- as shown above it's relatively trivial to do everywhere else.  For Windows, we can just leave this one blank (except for the current process, which does have its own function available in the API).
